local-cluster/src/cluster_tests.rs

```diff
@@ -527,20 +532,39 @@ pub fn start_gossip_voter(
                 });
 
                 for (parsed_vote, leader_vote_tx) in &parsed_vote_iter {
-                    if let Some(latest_vote_slot) = parsed_vote.last_voted_slot() {
+                    if let Some(vote_slot) = parsed_vote.last_voted_slot() {
+                        latest_vote_slot = vote_slot;
```

Comment:
For this to work you have to rely on the `parsed_vote_iter` being sorted right? For clarity I think we could just set this right after the `sort_by`, set it to the last element in the vector